iothub/service/src/Common/Amqp/ClientWebSocketTransport.cs

```diff
@@ -255,24 +258,22 @@ private static void HandleReadComplete(IAsyncResult result)
         private static bool ReadTaskDone(Task<int> taskResult, TransportAsyncCallbackArgs args)
         {
             IAsyncResult result = taskResult;
-            args.BytesTransfered = 0;  // reset bytes transferred
+            args.BytesTransfered = 0; // reset bytes transferred
+
             if (taskResult.IsFaulted)
             {
                 args.Exception = taskResult.Exception;
                 return true;
             }
-            else if (taskResult.IsCompleted)
+
+            if (taskResult.IsCompleted)
             {
                 args.BytesTransfered = taskResult.Result;
                 args.CompletedSynchronously = result.CompletedSynchronously;
                 return true;
```

Comment:
Are these returns needed? return taskResult.IsCanceled should be sufficient right?